File: mobile/lib/features/inventory/data/inventory_model.dart
class InventoryMovementModel {
  final String id;
  final String productId;
  final String type; // IN, OUT, ADJUSTMENT
  final int quantityChange;
  final String reason;
  final Map<String, dynamic> adjustedBy;
  final DateTime createdAt;

  InventoryMovementModel({
    required this.id,
    required this.productId,
    required this.type,
    required this.quantityChange,
    required this.reason,
    required this.adjustedBy,
    required this.createdAt,
  });

  factory InventoryMovementModel.fromJson(Map<String, dynamic> json) {
    return InventoryMovementModel(
      id: json['_id'] ?? json['id'] ?? '',
      productId: json['productId'] ?? '',
      type: json['type'] ?? '',
      quantityChange: (json['quantityChange'] as num?)?.toInt() ?? 0,
      reason: json['reason'] ?? '',
      adjustedBy: json['adjustedBy'] ?? {},
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}

File: server/src/modules/inventory/inventory.model.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IInventoryMovement extends Document {
  productId: mongoose.Types.ObjectId;
  type: 'IN' | 'OUT' | 'ADJUSTMENT';
  quantityChange: number;
  reason: string;
  adjustedBy: mongoose.Types.ObjectId;
  createdAt: Date;
  updatedAt: Date;
}

const inventoryMovementSchema = new Schema(
  {
    productId: { type: Schema.Types.ObjectId, ref: 'Product', required: true },
    type: { type: String, enum: ['IN', 'OUT', 'ADJUSTMENT'], required: true },
    quantityChange: { type: Number, required: true }, // can be positive or negative
    reason: { type: String, required: true },
    adjustedBy: { type: Schema.Types.ObjectId, ref: 'User', required: true },
  },
  { timestamps: true }
);

inventoryMovementSchema.index({ productId: 1, createdAt: -1 });

export const InventoryMovementModel = mongoose.model<IInventoryMovement>('InventoryMovement', inventoryMovementSchema);
